src/cmd/ls.rs

```diff
@@ -373,6 +373,10 @@ pub fn run_cmd<W: Write>(mut w: &mut W, cwd: &Path, cmd: &Cmd, project: &Project
                 if !ui::show_artifact(&name, art, &pat_case, &cmd.search_settings) {
                     continue;
                 }
+                if !match_made && !fmt_set.long && cmd.ty == OutType::List {
```

Comment:
instead of doing it this way, just make the `if` block:

```
if !names.is_empty() && !fmt_set.long && cmd.ty == OutType::List
```

without moving the if block.